src/core/util/config.cc

```diff
@@ -158,10 +158,14 @@ void Configuration::ParseConfig()
   bpo::store(
       bpo::command_line_parser(m_Args).options(cli_options).run(), m_Map);
   bpo::notify(m_Map);
+
+  // TODO(anonimal): Initializing of sources/streams/sinks must come after we've properly configured the logger.
+  //   we do this here so we can catch debug logging before instance "initialization". This is not ideal
+  core::SetupLogging(m_Map);
```

Comment:
Oops! Slight problem: now log options won't be set by config file. My fault! I should've looked before commenting.

We have two options now:
1. Setup logging after we parse config file *then* run help. This is a waste of cycles when running help.
2. Revert all this work and keep your original PR (minus the comment and call to SetupLogging). If we don't initialize custom logging, boost.log will use perfectly fine defaults. So, unless you want a colorful help section, just use boost.log defaults by calling LOG without SetupLogging.

Does that make sense?

Reply:
I changed it back to the 2. option.